x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomLink/CustomLinkFlyout/index.tsx

```diff
@@ -31,24 +31,25 @@ interface Props {
   customLinkSelected?: CustomLink;
   onSave: () => void;
   onDelete: () => void;
+  filters?: Filter[];
 }
 
 export const CustomLinkFlyout = ({
   onClose,
   customLinkSelected,
   onSave,
-  onDelete
+  onDelete,
+  filters
 }: Props) => {
   const { toasts } = useApmPluginContext().core.notifications;
   const [isSaving, setIsSaving] = useState(false);
 
   const [label, setLabel] = useState(customLinkSelected?.label || '');
   const [url, setUrl] = useState(customLinkSelected?.url || '');
-  const selectedFilters = customLinkSelected?.filters;
-  const [filters, setFilters] = useState(
-    selectedFilters?.length
-      ? selectedFilters
-      : ([{ key: '', value: '' }] as Filter[])
+  const _filters = customLinkSelected?.filters || filters;
+  const filtersEmptyState: Filter[] = [{ key: '', value: '' }];
+  const [selectedFilters, setSelectedFilters] = useState(
+    _filters?.length ? _filters : filtersEmptyState
```

Comment:
Hmm I'm not sure if this makes it more clear. I particularly don't see why `filters` should be an exception (to `url` and `label`). What do you think of adding a `defaults` prop (`{ url?:string; label?:string; filters?:Filter[] }` and a `customLinkId` prop? I think `defaults` is a better indication of what the prop actually does (which is, set some defaults), and you shouldn't have to use any type casts.

Reply:
I don't want to make `url` and `label` optional on CustomLink type, because, in fact, they will never be. A Custom Link will always have url, label and maybe filters.
 I added the `filters` props because in the end it's not a CustomLink, it's just possible filters. I can try to give it better names and refactor a bit this part, WDYT?

Reply:
I don't mean making them optional on the `CustomLink` type, just in the component. There's already a fallback here (`|| ''`) so making it optional (again just in the component) shouldn't change anything, other than making it easier to consume the component. So just to clarify, these would be the props (truncated)

```ts
interface Props {
	defaults: {
		url?:string;
    label?:string;
    filters?:Filter[]
  };
  customLinkId?:string;
}
```